canvas_grab/snapshot/canvas_file_snapshot.py

```diff
@@ -64,6 +64,8 @@ def yield_take_snapshot(self):
         folders = request_batcher.get_folders()
 
         for _, file in files.items():
+            file.display_name = re.sub(file_regex, '_', file.display_name)
```

Comment:
Not a good idea to directly modify objects from Canvas API. I would suggest doing normalization just before using them. e.g. 

```python
folder = folders[file.folder_id].full_name + "/"
# change to
folder = normalize_path(folders[file.folder_id].full_name) + "/"
```

```python
filename = normalize_path(filename, file_regex)
```

```python
key = f'pages/{normalize_path(page.title, file_regex)}.html'
```

Reply:
I get it, thank you :D